sdk/core/azure-core/inc/azure/core/cryptography/key_encryption_key.hpp

```diff
@@ -0,0 +1,57 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+/**
+ * @file
+ * @brief A key which is used to encrypt, or wrap, another key.
+ */
+
+#pragma once
+
+#include <azure/core/context.hpp>
+
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Core { namespace Cryptography {
+
+  /**
+   * @brief A key which is used to encrypt, or wrap, another key.
+   *
+   */
+  struct IKeyEncryptionKey
+  {
+    /**
+     * @brief The Id of the key used to perform cryptographic operations for the client.
+     *
+     */
+    std::string KeyId;
+
+    /**
+     * @brief Encrypts the specified key using the specified algorithm.
+     *
+     * @param algorithm The key wrap algorithm used to encrypt the specified key.
+     * @param key The key to be encrypted.
+     * @param context A #Azure::Core::Context controlling the request lifetime.
+     * @return The encrypted key bytes.
+     */
+    virtual std::vector<uint8_t> WrapKey(
+        std::string const& algorithm,
+        std::vector<uint8_t> const& key,
```

Comment:
What input values are considered invalid here? Is an empty key vector allowed? How large can the key vector be? I would expect some of that to be documented.